metrics/cli/docs/features/doctor_output_improvements/06_doctor_output_improvements.md

```diff
@@ -118,3 +139,166 @@ The validation output classes and models are a part of the Metrics project and s
+
+##### Making things work <a href="#ci-integrations-making-things-work" id="ci-integrations-making-things-work"></a>
+
+Consider the following steps needed to update the `CI Integrations Validate` command:
+1. Create the main abstractions in the `Validation` package: `ValidationTarget`, `ValidationConclusion`, `TargetValidationResult`, `ValidationResult`, `ValidationResultBuilder`, `ValidationResultPrinter`.
+2. Update the `ConfigFieldValidationConclusion` and `CoolIntegrationSourceValidationTarget` to use the `ValidationConclusion` and `ValidationTarget` respectively.
+3. Update the validation methods of each integration's validation delegate to return the `TargetValidationResult`.
+4. Update the `.validate()` methods of each validator to return the updated `ValidationResult` containing the `ValidationTarget`s and `TargetValidationResult`s.
+5. Delete the outdated abstractions covered by the `Validation` package: `ValidationTarget`, `ValidationConclusion`, `TargetValidationResult`, `ValidationResult`, `ValidationResultBuilder`, `ValidationResultPrinter`.
+
+Consider the following diagrams that demonstrate the updated config validation for the `CoolIntegration`:
+
+- Class diagram:
+  ![CI integrations validator class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/doctor_output_design/metrics/cli/docs/features/doctor_output_improvements/diagrams/ci_integrations_validator_class_diagram.puml)
+
+- Sequence diagram:
+  ![CI integrations validator sequence diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/doctor_output_design/metrics/cli/docs/features/doctor_output_improvements/diagrams/ci_integrations_validator_sequence_diagram.puml)
+
+#### Update the `Metrics CLI Doctor` command
+
+This subsection describes the enhancements needed to implement in the `Metrics CLI` tool to improve the `doctor` command output.
+
+##### DoctorCommand
+
+The `DoctorCommand` is the Metrics CLI command that verifies all required 3-rd party tools' versions against the [list of recommended versions](https://github.com/Flank/flank-dashboard/blob/master/metrics/cli/recommended_versions.yaml).
+
+In the scope of this feature, we need to update the `DoctorCommand` class to be responsible for printing the [`ValidationResult`](#validationresult) using the [`ValidationResultPrinter`](#validationresultprinter).
+
+##### Doctor
+
+The `Doctor` is a class used to check whether all required third-party CLIs are installed and get their versions. This class encapsulates the logic of the `DoctorCommand` and interacts with the 3-rd party services.
+
+We need to update the `.checkVersions()` method of the `Doctor` class to return the `ValidationResult` that will be printed by the `DoctorCommand`.
+
+The `ValidationResult` contains the `TargetValidationResult`s returned by the `.checkVersion()` methods of each 3-rd party service.
```

Comment:
🏷️ 
```suggestion
The `ValidationResult` contains the `TargetValidationResult`s for each 3-rd party service.
```
It seems like `InfoService.checkVersion()` method does not return a `TargetValidationResult`.